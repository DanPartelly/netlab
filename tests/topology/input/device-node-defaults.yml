#
# BGP address families -- full definition should only be present on SR OS
#

provider: clab

nodes: 
  h1:
    device: linux
  h2:
    device: linux
  rtr:
    device: cumulus
    id: 1

links:
- h1:
  rtr:
  type: lan
- h2:
  rtr:
  type: lan
