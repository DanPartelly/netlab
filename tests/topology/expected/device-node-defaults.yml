input:
- topology/input/device-node-defaults.yml
- package:topology-defaults.yml
links:
- bridge: input_1
  interfaces:
  - ipv4: 172.16.0.2/24
    node: h1
  - ipv4: 172.16.0.1/24
    node: rtr
  linkindex: 1
  node_count: 2
  prefix:
    ipv4: 172.16.0.0/24
  type: lan
- bridge: input_2
  interfaces:
  - ipv4: 172.16.1.3/24
    node: h2
  - ipv4: 172.16.1.1/24
    node: rtr
  linkindex: 2
  node_count: 2
  prefix:
    ipv4: 172.16.1.0/24
  type: lan
name: input
nodes:
  h1:
    box: python:3.9-alpine
    clab:
      kind: linux
    device: linux
    hostname: clab-input-h1
    id: 2
    interfaces:
    - bridge: input_1
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.0.2/24
      linkindex: 1
      name: h1 -> [rtr]
      neighbors:
      - ifname: swp1
        ipv4: 172.16.0.1/24
        node: rtr
      type: lan
    loopback:
      ipv4: 10.0.0.2/32
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.102
      mac: 08-4F-A9-00-00-02
    name: h1
    role: host
  h2:
    box: python:3.9-alpine
    clab:
      kind: linux
    device: linux
    hostname: clab-input-h2
    id: 3
    interfaces:
    - bridge: input_2
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.1.3/24
      linkindex: 2
      name: h2 -> [rtr]
      neighbors:
      - ifname: swp2
        ipv4: 172.16.1.1/24
        node: rtr
      type: lan
    loopback:
      ipv4: 10.0.0.3/32
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.103
      mac: 08-4F-A9-00-00-03
    name: h2
    role: host
  rtr:
    box: networkop/cx:4.3.0
    clab:
      kind: cvx
    device: cumulus
    hostname: clab-input-rtr
    id: 1
    interfaces:
    - bridge: input_1
      ifindex: 1
      ifname: swp1
      ipv4: 172.16.0.1/24
      linkindex: 1
      name: rtr -> [h1]
      neighbors:
      - ifname: eth1
        ipv4: 172.16.0.2/24
        node: h1
      type: lan
    - bridge: input_2
      ifindex: 2
      ifname: swp2
      ipv4: 172.16.1.1/24
      linkindex: 2
      name: rtr -> [h2]
      neighbors:
      - ifname: eth1
        ipv4: 172.16.1.3/24
        node: h2
      type: lan
    loopback:
      ipv4: 10.0.0.1/32
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.101
      mac: 08-4F-A9-00-00-01
    name: rtr
provider: clab
